src/main/java/Duke.java

```diff
@@ -1,10 +1,290 @@
+    }
+    
+    private Duke(Duke oldDuke) {
+        this.taskList = oldDuke.taskList.stream()
+                .collect(Collectors.toList());
+       
+        this.taskInfoList = oldDuke.taskInfoList.stream()
+                .collect(Collectors.toList());
+    }
+
+    public Duke() {
+        this.taskList = new ArrayList<>();
+        this.taskInfoList = new ArrayList<>();
+    }
+    
+    public Duke runCommand(String input) throws DukeException {
+        String[] argsArray = input.split(" ");
+        String command = argsArray[0];
+        
+        Duke tempDuke = this;
+        switch(command) {
+        case "todo": 
+            if (argsArray.length == 1) {
+                throw new DukeException("☹ OOPS!!! The description "
+                        + "of a todo cannot be empty.");
+            }
+            
+            tempDuke = this.processToDosInput(input).addToDos();
+            Task.taskCounter++;
+            break;
+        case "deadline":
+            if (argsArray.length == 1) {
+                throw new DukeException("☹ OOPS!!! The description "
+                        + "of a deadline cannot be empty.");
+            }
+            tempDuke = this.processDeadlinesInput(input).addDeadlines();
+            Task.taskCounter++;
+            break;
+        case "event":
+            if (argsArray.length == 1) {
+                throw new DukeException("☹ OOPS!!! The description "
+                        + "of an event cannot be empty.");
+            }
+            tempDuke = this.processEventsInput(input).addEvents();
+            Task.taskCounter++;
+            break;
+        case "done": 
+            tempDuke = this.completeTask(Integer.parseInt(argsArray[1]));
+            break;
+        case "list":
+            this.printList();
+            break;
+        case "bye": 
+            this.printOutput("Bye. Hope to see you again soon!");
+            break;
+        default:
+            throw new DukeException("☹ OOPS!!! I'm sorry, "
+                    + "but I don't know what that means :-(");                 
+        }
+
+        return new Duke(tempDuke);
+    }
+    
+    public Duke processEventsInput(String input) {
+        String[] argsArray = input.split(" ");
+        int splitIndex = 0;
+
+        splitIndex = IntStream.range(0, argsArray.length)
+                .filter(i -> (argsArray[i].equals("/at")))
+                .reduce(0, (x, y) -> x + y);
+
+        String taskName = "";
+        for (int j = 1; j < splitIndex; j++) {
+            if (j != 1) {
+                taskName += (" " + argsArray[j]);
+            } else {
+                taskName += (argsArray[j]);
+            }
+        }
+
+        String dateTime = "";
+        for (int k = splitIndex + 1; k < argsArray.length; k++) {
+            if (k > splitIndex + 1) {
+                dateTime += (" " + argsArray[k]);
+            } else {
+                dateTime += (argsArray[k]);
+            }
+        }
+

```

Comment:
Maybe can write comments to explain what this does, or have another method to encapsulate this for-loop (for SLAP)?

Reply:
for taskName and dateTime

Reply:
This is a good suggestion. I am refactoring my entire project so i'll keep this in mind.